environment/environment.go

```diff
@@ -31,15 +31,15 @@ type ServiceDetails struct {
 	LocalURL  *url.URL
 }
 
-// GetExplorerMockClient will return all instantiated Explorer mock client for a given environment
-func GetExplorerMockClient(env Environment) (*client.ExplorerClient, error) {
-	sd, err := env.GetServiceDetails(ExplorerWSPort)
+func GetExplorerClient(getServiceDetails func (remotePort uint16) (*ServiceDetails, error)) (*client.ExplorerClient, error) {
```

Comment:
Don't pass a func in here, rather have two functions:

`NewExplorerClient(localURL string)`
`GetExplorerClient(env Environment)`

With `GetExplorerClient` calling `NewExplorerClient` with the local URL.